server/src/main/java/org/elasticsearch/action/admin/indices/settings/put/UpdateSettingsRequest.java

```diff
@@ -239,19 +239,22 @@ public String toString() {
 
     @Override
     public boolean equals(Object o) {
-        if (super.equals(o)) {
-            UpdateSettingsRequest that = (UpdateSettingsRequest) o;
-            return Objects.equals(settings, that.settings)
-                    && Objects.equals(indicesOptions, that.indicesOptions)
-                    && Objects.equals(preserveExisting, that.preserveExisting)
-                    && Arrays.equals(indices, that.indices);
+        if (this == o) {
+            return true;
         }
-        return false;
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+        UpdateSettingsRequest that = (UpdateSettingsRequest) o;
+        return Objects.equals(settings, that.settings)
+                && Objects.equals(indicesOptions, that.indicesOptions)
+                && Objects.equals(preserveExisting, that.preserveExisting)
+                && Arrays.equals(indices, that.indices);
```

Comment:
don't you need to take also timeout and masterNodeTimeout into account?

Reply:
I would suspect proper equals/hashcode tests to fail currently, maybe also tests need to be updated to mutate such fields.

Reply:
Now the equals and the hashCode use all request fields except for the flatSettings ( as the flatSettings are not serialized ).